ansible/playbooks/roles/upgrade/tasks/postgres-exporter.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: Check current version
+  shell: /opt/postgres_exporter/postgres_exporter --version 2>&1 | grep postgres_exporter | awk '{print $3}'
+  register: postgres_exporter_current_version
+
+- name: Postgres Exporter | Include defaults from postgres_exporter role
+  include_vars:
+    file: roles/postgres_exporter/defaults/main.yml
+    name: exporter_defaults
+
+- name: Ugrade postgres_exporter
+  include_role:
+    name: postgres_exporter
+    tasks_from: upgrade
+  # It looks strange but old and new version has different response syntax for that command :(
+  when:
+  - (postgres_exporter_current_version.stdout == "(built") or (postgres_exporter_current_version.stdout is version (exporter_defaults.exporter.version, '<'))
```

Comment:
Up to v0.9.0 the output from the command `postgres_exporter --version` was always:
```
user@test:/tmp/postgres_exporter/postgres_exporter-0.9.0.linux-amd64$ postgres_exporter --version
postgres_exporter 0.0.1-rev (built with go1.15.8)
```
It's been fixed in v0.10.0 https://github.com/prometheus-community/postgres_exporter/pull/494/files#diff-f1f9312de66d6e1570004277f9fc74e50ac9301aea667950f9326d8f7b4b8d6fL59.

I am convinced that we should parse output here, even extract the version `0.0.1`, and not use the command `postgres_exporter --version 2>&1 | grep postgres_exporter | awk '{print $3}'` which returns `(built` as output.

```
user@test:/tmp/postgres_exporter/postgres_exporter-0.9.0.linux-amd64$ postgres_exporter --version 2>&1 | grep postgres_exporter | awk '{print $3}'
(built
```

Reply:
the issue is that old and new version return version separated by spaces in different column. So the same command on new version of exporter returns you proper version and the old one such string.
So I assume we should follow new structure and leave this ugly workaround forever to keep backward compability.
Thats I leave the commend above but maybe it is not clear enough or you have different idea what to do with it

Reply:
Please check if now you are fine with the version verification task :)